pkg/objectset/objectset.go

```diff
@@ -167,6 +167,31 @@ func (o *ObjectSet) GVKOrder(known ...schema.GroupVersionKind) []schema.GroupVer
 	return append(o.gvkOrder, rest...)
 }
 
+// Namespaces all distinct namespaces found on the objects in this set.
+func (o *ObjectSet) Namespaces() (namespaces []string) {
+	for _, _gvk := range o.ObjectsByGVK() {
```

Comment:
Any specific reason it's `_gvk` and not just `_gvk`? In fact `gvk` is probably not the best name either as GVK is key in `ObjectsByGVK` and value is `map[ObjectKey]runtime.Object` - so `objsByKey` is probably more descriptive.